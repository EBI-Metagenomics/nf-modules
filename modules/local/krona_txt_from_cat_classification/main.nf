process KRONA_TXT_FROM_CAT_CLASSIFICATION {
    tag "${meta.id}"
    label 'process_single'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'oras://community.wave.seqera.io/library/python:3.13.1--9856f872fdeac74e':
        'community.wave.seqera.io/library/python:3.13.1--d00663700fcc8bcf' }"

    input:
    tuple val(meta), path(cat_output)
    tuple val(meta2), path(taxonomy)

    output:
    tuple val(meta), path("${meta.id}.krona.txt"), emit: krona_txt
    path "versions.yml"                          , emit: versions

    script:
    """
    krona_txt_from_cat_classification.py \\
        --input ${cat_output} \\
        --output ${meta.id}.krona.txt \\
        --names_dmp ${taxonomy}/names.dmp \\
        --nodes_dmp ${taxonomy}/nodes.dmp


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version 2>&1 | sed 's/Python //g')
    END_VERSIONS
    """

    stub:
    """
    touch ${meta.id}.krona.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version 2>&1 | sed 's/Python //g')
    END_VERSIONS
    """
}