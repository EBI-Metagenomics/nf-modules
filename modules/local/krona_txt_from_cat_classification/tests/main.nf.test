nextflow_process {

    name "Test Process KRONA_TXT_FROM_CAT_CLASSIFICATION"
    script "../main.nf"
    process "KRONA_TXT_FROM_CAT_CLASSIFICATION"

    tag "modules"
    tag "modules_nfcore"
    tag "modules_ebimetagenomics"

    test("Run KRONA_TXT_FROM_CAT_CLASSIFICATION module - test dataset") {

        when {
            process {
                """
                input[0] = [ [id:'test'], [ file("${moduleDir}/tests/data/cat.tsv", checkIfExists: true) ] ]
                input[1] = [ [id:'test_db'], [ file("${moduleDir}/tests/data/tax", checkIfExists: true) ] ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("Run KRONA_TXT_FROM_CAT_CLASSIFICATION module - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [ [id:'test'], [ file("${moduleDir}/tests/data/cat.tsv", checkIfExists: true) ] ]
                input[1] = [ [id:'test_db'], [ file("${moduleDir}/tests/data/tax", checkIfExists: true) ] ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
