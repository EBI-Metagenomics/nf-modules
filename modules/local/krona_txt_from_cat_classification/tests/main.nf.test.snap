{
    "Run KRONA_TXT_FROM_CAT_CLASSIFICATION module - test dataset": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.krona.txt:md5,62e1bcaac47b979cafb7364cdc48eafa"
                    ]
                ],
                "1": [
                    "versions.yml:md5,8f885e77b481995723e937c8ed106642"
                ],
                "krona_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.krona.txt:md5,62e1bcaac47b979cafb7364cdc48eafa"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8f885e77b481995723e937c8ed106642"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-03-24T10:55:07.554155"
    },
    "Run KRONA_TXT_FROM_CAT_CLASSIFICATION module - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.krona.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,8f885e77b481995723e937c8ed106642"
                ],
                "krona_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.krona.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8f885e77b481995723e937c8ed106642"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-03-24T10:56:06.833574"
    }
}